job/src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowRun.java

```diff
@@ -272,6 +292,50 @@ private void copyLogs() {
             }
         }
     }
+
+    private String getLogPrefix(FlowNode node) {
+        BodyInvocationAction bodyInvocationAction = node.getAction(BodyInvocationAction.class);
+
+        if (bodyInvocationAction != null) {
+            BodyExecutionLabelAction bodyLabelAction = node.getAction(BodyExecutionLabelAction.class);
+
+            if (bodyLabelAction != null) {
+                return bodyLabelAction.getBodyLabel();
+            } else {
+                // This exec body does not have a body label. Stop now.
+                return null;
+            }
+        }
```

Comment:
This seems fishy. What if there is a block like `node {}` inside `parallel`? IIUC this code would fail to display any branch prefix for steps run inside that.

I do not see any particular reason to check for `BodyInvocationAction` at all.